### PR TITLE
Custom 404 page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Implement custom 404 page.

Update button component to support responsive size prop.

Update layout with menu component to use "overlay" setting on non-CMS pages.

Refactor prodcut recommendation list to be able to reuse it on 404 page.

## Justify why they are needed

To have a custom 404 page.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1630** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1630" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1630?utm_source=stack-comment).

File: packages/ui/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -60,51 +55,33 @@ Button.displayName = 'Button'
...
+    '&:disabled': {
+      cursor: 'default',
+    },
+
+    '&:focus-visible': {
+      boxShadow: `0 0 0 2px ${theme.colors.textPrimary}`,
+    },
   },
-}))
+  ({ size = 'large' }) => getButtonSizeStyles(size),
```
Comment (alebedev) [inline packages/ui/src/components/Button/Button.tsx:83]: Nice!

File: packages/ui/src/components/Button/Button.helpers.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+type ButtonSizeVariant = 'small' | 'medium' | 'large'
+
+export type ButtonSize = ButtonSizeVariant | Partial<Record<Level | 'base', ButtonSizeVariant>>
+
+export const getButtonSizeStyles = (size: ButtonSize) => {
+  if (typeof size !== 'object') {
+    return SIZE_STYLES[size]
+  }
+
+  const styles = Object.entries(size).reduce((acc, [level, variant]) => {
```
Comment (alebedev) [inline packages/ui/src/components/Button/Button.helpers.ts:20]: Nit: let's extract a util for responsive sizes.  Here we use `base`, in `Text` we use `_`, can we use the same name everywhere?

File: packages/ui/src/components/Button/Button.helpers.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    if (level === 'base') return { ...acc, ...SIZE_STYLES[variant] }
+
+    return {
+      ...acc,
```
Comment (alebedev) [inline packages/ui/src/components/Button/Button.helpers.ts:24]: Nit: copy on every iteration, `forEach` with Object.assign would be better for performance and cleaner

File: apps/store/src/components/ProductRecommendationList/useProductRecommendations.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { useProductRecommendationsQuery } from '@/services/apollo/generated'
+import { useShopSession } from '@/services/shopSession/ShopSessionContext'
+
+export const useProductRecommendations = () => {
+  const { shopSession } = useShopSession()
+
+  const shopSessionId = typeof shopSession?.id === 'string' ? shopSession.id : undefined
```
Comment (alebedev) [inline apps/store/src/components/ProductRecommendationList/useProductRecommendations.ts:7]: NIt: `shopSession?.id ?? ''`

File: apps/store/src/components/FourOhFourPage/FourOhFourPage.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+            variant="primary"
+            size={{ base: 'small', lg: 'medium' }}
+            href={PageLink.home()}
+          >
+            {t('404_PAGE_BUTTON')}
+          </ButtonNextLink>
+        </SpaceFlex>
+      </Wrapper>
+
+      {productRecommendations && (
```
Comment (alebedev) [inline apps/store/src/components/FourOhFourPage/FourOhFourPage.tsx:34]: Should we check `.length > 0` too?

**Review by alebedev — APPROVED** [review]
Looks good, some code style questions

File: packages/ui/src/components/Button/Button.helpers.ts (hunk 3 above)
Comment (robinandeer) [reply packages/ui/src/components/Button/Button.helpers.ts:24]: will do when extracting the utility

File: apps/store/src/components/ProductRecommendationList/useProductRecommendations.ts (hunk 4 above)
Comment (robinandeer) [reply apps/store/src/components/ProductRecommendationList/useProductRecommendations.ts:7]: yeah but ID has type any from the API and if I do this then TS will keep that type which is not what we want :P

**Pull request merged**